runtime/shared_common/CacheMap.cpp

```diff
@@ -883,39 +883,29 @@ SH_CacheMap::createNewSegment(J9VMThread* currentThread, UDATA type, J9MemorySeg
 void
 SH_CacheMap::updateROMSegmentList(J9VMThread* currentThread, bool hasClassSegmentMutex)
 {
-	SH_CompositeCacheImpl* cache;
+	SH_CompositeCacheImpl* cache = _ccHead;
 #if defined(J9VM_THR_PREEMPTIVE)
-	J9JavaVM* vm = currentThread->javaVM;
-
-	omrthread_monitor_t classSegmentMutex = vm->classMemorySegments->segmentMutex;
+	omrthread_monitor_t classSegmentMutex = currentThread->javaVM->classMemorySegments->segmentMutex;
```

Comment:
The initialization here can move into the `if (!hasClassSegmentMutex) {` block below.

Reply:
classSegmentMutex is also used when exiting the mutex and in the trace Trc_SHR_Assert_ShouldHaveLocalMutex which are outside of the `if (!hasClassSegmentMutex) { `block

Reply:
Everything was address except this comment.

Reply:
Ok, I see.